drivers/sx126x/sx126x_radio_hal.c

```diff
@@ -0,0 +1,668 @@
+    sx126x_get_lora_pkt_status(dev, &pkt_status);
+    if (info) {
+        info->lqi = pkt_status.snr_pkt_in_db;
+        info->rssi = pkt_status.rssi_pkt_in_dbm;
+    }
+
+      /* Put PSDU to the output buffer */
+    
+    if (buf == NULL) {
+        return rx_buffer_status.pld_len_in_bytes;
+    }
+
+    if (rx_buffer_status.pld_len_in_bytes > max_size) {
+        return -ENOBUFS;
+    }
+
+    if (rx_buffer_status.pld_len_in_bytes < 3) {
+        return -EBADMSG;
+    }
+    sx126x_read_buffer(dev, rx_buffer_status.buffer_start_pointer, (uint8_t*)buf, rx_buffer_status.pld_len_in_bytes);
+    DEBUG("[sx126x] first 3 bytes of received packet: %d %d %d\n", *(uint8_t*)buf, *((uint8_t*)buf+1), *((uint8_t*)buf+2));
+    return rx_buffer_status.pld_len_in_bytes;
+}
+
+static int _set_cca_threshold(ieee802154_dev_t *hal, int8_t threshold)
+{
+    (void)hal;
+    (void)threshold;
+
+    return 0;
+}
+
+static int _config_phy(ieee802154_dev_t *hal, const ieee802154_phy_conf_t *conf){
+    (void)hal;
+    (void)conf;
+    sx126x_t *dev = hal->priv;
+    uint8_t channel = conf->channel;
+    int8_t pow = conf->pow;
+    if((channel > SX126X_CHAN_MAX))
+        return -EINVAL;
+    
+    sx126x_set_channel(dev, (channel)*SX126X_HAL_CHAN_SPACING + SX126X_HAL_CHAN_BASE);
+    if((pow < SX126X_POWER_MIN) && (pow > SX126X_POWER_MAX))
+        return -EINVAL;
+
+    sx126x_set_tx_params(dev, pow, SX126X_RAMP_10_US);
+    return 0;
+}
+
+static int _off(ieee802154_dev_t *hal)
+{
+    (void)hal;
+    sx126x_t *dev = hal->priv;
+    sx126x_set_sleep(dev, SX126X_SLEEP_CFG_COLD_START);
+    return 0;
+}
+
+static int _config_addr_filter(ieee802154_dev_t *hal, ieee802154_af_cmd_t cmd, const void *value)
+{
+    (void)cmd;
+    (void)value;
+    uint16_t pan_id = *(uint16_t*)value;
+    sx126x_t *dev = hal->priv;
+    switch(cmd) {
+        case IEEE802154_AF_SHORT_ADDR:
+            memcpy(dev->short_addr, value, IEEE802154_SHORT_ADDRESS_LEN);
+            break;
+        case IEEE802154_AF_EXT_ADDR:
+            memcpy(dev->long_addr, value, IEEE802154_LONG_ADDRESS_LEN);
+            break;
+        case IEEE802154_AF_PANID:
+            dev->pan_id = pan_id;
+            break;
+        case IEEE802154_AF_PAN_COORD:
+            return -ENOTSUP;
+    }
+    return 0;
+}
+
+static int _request_on(ieee802154_dev_t *hal)
+{
+    (void)hal;
+    sx126x_t *dev = hal->priv;
+    _set_state(dev, STATE_IDLE);
+    return 0;
+}
+
+static int _confirm_on(ieee802154_dev_t *hal)
+{
+    (void)hal;
+
+    return -ENOTSUP;
+}
+
+static int _set_cca_mode(ieee802154_dev_t *hal, ieee802154_cca_mode_t mode)
+{
+    (void)hal;
+    sx126x_t* dev = hal->priv;
+    DEBUG("[sx126x] netdev: set_cca_mode \n");
+        dev->cad_params.cad_exit_mode = SX126X_CAD_ONLY,
+        dev->cad_params.cad_detect_min = 10,
+        dev->cad_params.cad_detect_peak = 22,
+        dev->cad_params.cad_symb_nb = SX126X_CAD_02_SYMB,
+        dev->cad_params.cad_timeout = 0x000F00;
```

Comment:
is `cad_timeout` in symbols? If so, this value is too big. In case of IEEE 802.15.4, I would set it to 0x08